force-app/main/default/classes/DMLOperations.cls

```diff
@@ -161,14 +182,58 @@ public with sharing class DMLOperations {
 	public static Account upsertAccount(String accountName) {
 		// Query for the Account
 		List<Account> accounts = [SELECT Id FROM Account WHERE Name = :accountName];
+        Account acct = new Account();
+		if (!accounts.isempty()){
+            acct = accounts[0];
+            acct.Description = 'Updated Account';   // if the Account exists, update the description
+        } else {                                    // otherwise, create a new Account and set the description
+            acct.Name = accountName;
+            acct.Description = 'New Account';
+        }   
+		upsert acct;                        		// Upsert the Account
+		return acct;
+	}
+    
+    /**
+     * Utility 9.5
+     * Created to facilitate Question 10
+	 * Inserts a list of Accounts with specified names.
+	 * For each account in the list:
+        * If the Account with the given name already exists, update the description to "Updated Account".
+	    * If no such Account exists, it creates a new one and sets the description to "New Account".
+	 * @param accountName The name of the Account to be upserted.
+	 * @return The Account that was upserted.
+	 */
+	
+    public static Map<String, Id> insertAccounts(Set<String> acctNamesSet) {
```

Comment:
If you are truely overloading, it would always have to return the same type. In this case you have a different type and a different method name, so it not exactly overloading, but it is the same concept. If this were me, building for a production case, I would change question 9 to call this 9.5 method, with the single input from 9 into a set, and then extract the single returned out of the Map when returned. 

`public static Account upsertAccount(String accountName) {`
`    return new Account(Id = insertAccounts(new Set<String> { accountName }).get(accountName));`
`}
`